crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs

```diff
@@ -576,6 +576,7 @@ fn verify_soft_batch_signature<C: Context>(
     let unsigned = UnsignedSoftConfirmationBatch::new(
         soft_batch.da_slot_height(),
         soft_batch.da_slot_hash(),
+        soft_batch.da_slot_merkle_root(),
         soft_batch.pre_state_root(),
         soft_batch.txs(),
         soft_batch.l1_fee_rate(),
```

Comment:
in this file 

in begin_soft_batch function, observe how we assert values in soft confirmation against the ones full node itself supplied. 

da_slot_transaction_commitment should also be asserted that way.